File: LanQiaoCup/11thprovinceAB/f.cpp
/**
 *    author: subobo
 *    created: 31.03.2022 21:09:57
**/
#include <bits/stdc++.h>

using namespace std;

#ifdef LOCAL
#define debug(...) cerr << "[" << #__VA_ARGS__ << "]:", debug_out(__VA_ARGS__)
#else
#define debug(...) 42
#endif

int main() {
  ios::sync_with_stdio(false);
  cin.tie(0);
  int n;
  cin >> n;
  long long pref = 0;
  long long ans = 0;
  for (int i = 0; i < n; i++) {
    long long x;
    cin >> x;
    ans += pref * x;
    pref += x;
  }
  cout << ans << '\n';
  return 0;
}

File: CodeForces/r806_div4/f.py
# author: subobo
# created: 19.07.2022 10:43:12

tt = int(input())
for qq in range(tt):
  n = int(input())
  a = [0]
  a.extend(list(map(int, input().split())))
  
  pref = [0] * (n + 1)
  for i in range(1, n + 1):
    pref[i] = pref[i - 1] + int(a[i] < i)
  
  ans = 0
  for j in range(1, n + 1):
    if a[j] < j and a[j] - 1 >= 1:
      ans += pref[a[j] - 1]

  print(ans)